legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitlabWorkflowJobApi.java

```diff
@@ -0,0 +1,326 @@
+                        () -> "Error getting workflow job " + workflowJobId + " in " + getRefInfoForException());
+            }
+
+            if (!getRef().equals(job.getRef()))
+            {
+                throw new LegendSDLCServerException("Unknown workflow job in " + getRefInfoForException() + ": " + workflowJobId, Response.Status.NOT_FOUND);
+            }
+
+            if (!workflowId.equals(toStringIfNotNull(job.getPipeline().getId())))
+            {
+                throw new LegendSDLCServerException("Unknown workflow job in " + getRefInfoForException() + ", workflow " + workflowId + ": " + workflowJobId, Response.Status.NOT_FOUND);
+            }
+
+            return jobTrace;
+        }
+
+        protected abstract String getRef();
+
+        protected abstract String getRefInfoForException();
+    }
+
+    private static WorkflowJob fromGitLabJob(String projectId, Job job)
+    {
+        String id = toStringIfNotNull(job.getId());
+        String name = toStringIfNotNull(job.getName());
+        String revisionId = job.getPipeline().getSha();
+        String webURL = job.getWebUrl();
+        WorkflowJobStatus status = fromGitLabJobStatus(job.getStatus());
+        Instant createdAt = toInstantIfNotNull(job.getCreatedAt());
+        Instant startedAt = toInstantIfNotNull(job.getStartedAt());
+        Instant finishedAt = toInstantIfNotNull(job.getFinishedAt());
+        return new WorkflowJob()
+        {
+            @Override
+            public String getId()
+            {
+                return id;
+            }
+
+            @Override
+            public String getName()
+            {
+                return name;
+            }
+
+            @Override
+            public String getProjectId()
+            {
+                return projectId;
+            }
+
+            @Override
+            public String getRevisionId()
+            {
+                return revisionId;
+            }
+
+            @Override
+            public WorkflowJobStatus getStatus()
+            {
+                return status;
+            }
+
+            @Override
+            public Instant getCreatedAt()
+            {
+                return createdAt;
+            }
+
+            @Override
+            public Instant getStartedAt()
+            {
+                return startedAt;
+            }
+
+            @Override
+            public Instant getFinishedAt()
+            {
+                return finishedAt;
+            }
+
+            @Override
+            public String getWebURL()
+            {
+                return webURL;
+            }
+        };
+    }
+
+    private static WorkflowJobStatus fromGitLabJobStatus(JobStatus jobStatus)
+    {
+        if (jobStatus == null)
+        {
+            return null;
+        }
+        switch (jobStatus)
+        {
+            case CREATED:
+            case PENDING:
+            {
+                return WorkflowJobStatus.PENDING;
+            }
```

Comment:
As per my earlier comment on WorkflowJobStatus, we may want to reconsider some of the logic here.